viewer/src/component/three/RevealManager.ts

```diff
@@ -0,0 +1,291 @@
+import { filterCurrentWantedSectors } from '../../models/cad/filterCurrentWantedSectors';
+import { discardSector } from '../../views/threejs/cad/discardSector';
+import { CadLoadingHints } from '../../models/cad/CadLoadingHints';
+import { CadNode } from '../../views/threejs/cad/CadNode';
+import { ProximitySectorCuller } from '../../culling/ProximitySectorCuller';
+import { CadBudget } from '../../models/cad/CadBudget';
+import { ConsumedSector } from '../../data/model/ConsumedSector';
+import { ModelNodeAppearance } from '../../views/common/cad/ModelNodeAppearance';
+import { Sector, SectorQuads } from '../../models/cad/types';
+import { File3dFormat } from '../../data/model/File3dFormat';
+import { Repository } from '../../repository/cad/Repository';
+import { MaterialManager } from '../../views/threejs/cad/MaterialManager';
+import { Cad } from '../../data/model/Cad';
+import { createThreeJsPointCloudNode } from '../../views/threejs';
+import { PotreeGroupWrapper } from '../../views/threejs/pointcloud/PotreeGroupWrapper';
+import { PotreeNodeWrapper } from '../../views/threejs/pointcloud/PotreeNodeWrapper';
+import { PointCloud } from '../../data/model/PointCloud';
+import { SectorCuller } from '../../culling/SectorCuller';
+import { WantedSector } from '../../data/model/WantedSector';
+
+export interface RevealOptions {
+  nodeAppearance?: ModelNodeAppearance;
+  budget?: CadBudget;
+  // internal options are experimental and may change in the future
+  internal?: {
+    parseCallback?: (parsed: { lod: string; data: Sector | SectorQuads }) => void;
+    sectorCuller?: SectorCuller;
+  };
+}
+
+export type OnDataUpdated = () => void;
+
+export class RevealManager {
+  protected readonly _materialManager: MaterialManager;
+  private readonly _sectorRepository: Repository;
+  private readonly _modelSubject: Subject<Cad | PointCloud> = new Subject();
+  private readonly _loadingHintsSubject: Subject<CadLoadingHints> = new BehaviorSubject({});
+  private readonly _cameraSubject: Subject<THREE.PerspectiveCamera>;
+  private readonly _modelObservable: Observable<CadNode | [PotreeGroupWrapper, PotreeNodeWrapper]>;
+  private readonly _cadObservable: Observable<CadNode[]>;
+
+  private readonly _cadNodeMap: Map<string, CadNode> = new Map();
+
+  private _sectorCuller: SectorCuller;
+  // private _budget: CadBudget;
+
+  constructor(
+    client: CogniteClient,
+    sectorRepository: Repository,
+    materialManager: MaterialManager,
+    dataUpdatedCallback: OnDataUpdated,
+    options?: RevealOptions
+  ) {
+    this._sectorCuller = (options && options.internal && options.internal.sectorCuller) || new ProximitySectorCuller();
+    // this._budget = (options && options.budget) || createDefaultCadBudget();
+    this._sectorRepository = sectorRepository;
+    this._materialManager = materialManager;
+    this._modelObservable = this._modelSubject.pipe(
```

Comment:
Can pipelines be visualized with some simple ASCII-art inline in the code? Hard for me to follow what's going on